### PR TITLE
Fix metrics being mutated during report

### PR DESCRIPTION
**Pull request by michi88**
fixes https://github.com/wavefrontHQ/wavefront-sdk-python/issues/87

### CODE REVIEW

**Comment by michi88** [comment]
Also fixes the build now. Not sure what the intention of the setup.py command was, remove it.

File: wavefront_sdk/entities/histogram/histogram_impl.py [hunk 1]
```diff
@@ -155,17 +155,16 @@ def bulk_update_dist_by_thread_id(self, thread_id, means, counts):
     def get_centroids(self):
         """Get list of centroids for dists of all threads in this minute."""
         centroids = []
-        for thread_id in self.per_thread_dist:
-            centroids.extend(
-                self.per_thread_dist[thread_id].centroids_to_list())
+        for thread_item in self.per_thread_dist.items():
```
Comment (keep94) [inline wavefront_sdk/entities/histogram/histogram_impl.py:158]: consider using 
```
for _, thread_item in self.per_thread_dist.items():
    centroids.extend(thread_item.centroids_to_list())
```

File: wavefront_sdk/common/metrics/registry.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ def _schedule_timer(self):
     # pylint: disable=broad-except
     def _report(self, timeout_secs=None):
         timestamp = time.time()
-        for key, val in self.metrics.items():
+        for key, val in self.metrics.copy().items():
```
Comment (keep94) [inline wavefront_sdk/common/metrics/registry.py:44]: Consider adding a comment explaining why this change is necessary. Just explain that copying the dict here protects from concurrent modification while iterating.

**Review by keep94 — CHANGES_REQUESTED** [review]
Thank you for making all these changes.

**Comment by keep94** [comment]
Michi88, when you have a chance, would you check the box that says allow edits by maintainers?  I have already done the changes I asked for locally.  If you check that box, I can amend and merge your pull request.

**Comment by michi88** [comment]
@keep94 done!

File: wavefront_sdk/common/metrics/registry.py (hunk 2 above)
Comment (keep94) [reply wavefront_sdk/common/metrics/registry.py:46]: Done.

File: wavefront_sdk/entities/histogram/histogram_impl.py (hunk 1 above)
Comment (keep94) [reply wavefront_sdk/entities/histogram/histogram_impl.py]: Done. Used digest instead of thread item as the variable name.

**Review by keep94 — APPROVED** [review]

**Comment by keep94** [comment]
michi88,
Thank you for contributing to our project and fixing our build.

File: .travis.yml [hunk 3]
```diff
@@ -19,9 +19,7 @@ before_script:
   - python -m pydocstyle
 script:
   - python -m unittest discover
-  - if [[ "${TRAVIS_PULL_REQUEST}" != "false" ]]; then
```
Comment (ustinov) [inline .travis.yml:22]: @michi88 @keep94 folks please bring back the check, is was making sure the version in setup.py is incremented as needed in every PR that's being merged

**Review by ustinov — CHANGES_REQUESTED** [review]

File: wavefront_sdk/entities/histogram/histogram_impl.py [hunk 4]
```diff
@@ -155,17 +155,16 @@ def bulk_update_dist_by_thread_id(self, thread_id, means, counts):
     def get_centroids(self):
         """Get list of centroids for dists of all threads in this minute."""
         centroids = []
-        for thread_id in self.per_thread_dist:
-            centroids.extend(
-                self.per_thread_dist[thread_id].centroids_to_list())
+        for _, digest in self.per_thread_dist.items():
```
Comment (ustinov) [inline wavefront_sdk/entities/histogram/histogram_impl.py:158]: Please consider using the `.values()` view instead of `.items()` if you don't use keys

**Pull request merged**